watchdog_reader.go

```diff
@@ -23,9 +25,14 @@ func newWatchdogReader(reader io.Reader, timeout time.Duration, timer *time.Time
 
 // Read reads up to len(p) bytes into p
 func (t *watchdogReader) Read(p []byte) (n int, err error) {
-	// FIXME limit the amount of data read in one chunk so as to not exceed the timeout?
+	//never read more bytes than watchdogChunkSize
+	readTarget := p
+	if len(p) > watchdogChunkSize {
+		readTarget = p[0:watchdogChunkSize]
+	}
+
```

Comment:
I can see this limiting the performance though for the faster transfers with bigger buffers.

Could wrap the t.reader.Read in a loop and call resetTimer() each time through the loop so the watchdogReader.Read() call still fills p?